code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm

```diff
@@ -82,4 +82,241 @@
+	projectiletype = /obj/item/projectile/leaper
+	projectilesound = 'sound/weapons/pierce.ogg'
+	ranged_cooldown_time = 30
+	pixel_x = -16
+	pixel_y = -16
+	icon = 'icons/mob/jungle/arachnid.dmi'
+	icon_state = "leaper"
+	icon_living = "leaper"
+	icon_dead = "leaper_dead"
+	layer = LARGE_MOB_LAYER
+	name = "leaper"
+	desc = "Commonly referred to as 'leapers', the Geron Toad is a massive beast that spits out highly pressurized bubbles containing a unique toxin, knocking down its prey and then crushing it with its girth."
+	speed = 10
+	stat_attack = 1
+	robust_searching = 1
+	var/hopping = FALSE
+	var/hop_cooldown = 0 //Strictly for player controlled leapers
+
+/obj/item/projectile/leaper
+	name = "leaper bubble"
+	icon_state = "leaper"
+	knockdown = 50
+	damage = 0
+	range = 7
+	hitsound = 'sound/effects/snap.ogg'
+	nondirectional_sprite = TRUE
+	impact_effect_type = /obj/effect/temp_visual/leaper_projectile_impact
+
+/obj/item/projectile/leaper/on_hit(atom/target, blocked = 0)
+	..()
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		C.reagents.add_reagent("leaper_venom", 5)
+
+/obj/item/projectile/leaper/on_range()
+	var/turf/T = get_turf(src)
+	..()
+	new /obj/structure/leaper_bubble(T)
+
+/obj/effect/temp_visual/leaper_projectile_impact
+	name = "leaper bubble"
+	icon = 'icons/obj/projectiles.dmi'
+	icon_state = "leaper_bubble_pop"
+	layer = ABOVE_ALL_MOB_LAYER
+	duration = 3
+
+/obj/effect/temp_visual/leaper_projectile_impact/Initialize()
+	. = ..()
+	new /obj/effect/decal/cleanable/leaper_sludge(get_turf(src))
+
+/obj/effect/decal/cleanable/leaper_sludge
+	name = "leaper sludge"
+	desc = "A small pool of sludge, containing trace amounts of leaper venom"
+	icon = 'icons/effects/tomatodecal.dmi'
+	icon_state = "tomato_floor1"
+
+/obj/structure/leaper_bubble
+	name = "leaper bubble"
+	desc = "A floating bubble containing leaper venom, the contents are under a surprising amount of pressure."
+	icon = 'icons/obj/projectiles.dmi'
+	icon_state = "leaper"
+	obj_integrity = 10
+	max_integrity = 10
+	density = FALSE
+
+/obj/structure/leaper_bubble/Initialize()
+	. = ..()
+	float(on = TRUE)
+	QDEL_IN(src, 100)
+
+/obj/structure/leaper_bubble/Destroy()
+	new /obj/effect/temp_visual/leaper_projectile_impact(get_turf(src))
+	playsound(src,'sound/effects/snap.ogg',50, 1, -1)
+	..()
+
+/obj/structure/leaper_bubble/Crossed(atom/movable/AM)
+	if(isliving(AM))
+		var/mob/living/L = AM
+		if(!istype(L, /mob/living/simple_animal/hostile/jungle/leaper))
+			playsound(src,'sound/effects/snap.ogg',50, 1, -1)
+			L.Knockdown(50)
+			if(iscarbon(L))
+				var/mob/living/carbon/C = L
+				C.reagents.add_reagent("leaper_venom", 5)
+			qdel(src)
+	..()
+
+/datum/reagent/toxin/leaper_venom
+	name = "Leaper venom"
+	id = "leaper_venom"
+	description = "A toxin spat out by leapers that while harmless in small doses, quickly creates a toxic reaction if too much is in the body."
+	color = "#801E28" // rgb: 128, 30, 40
+	toxpwr = 0
+	taste_description = "frog"
```

Comment:
you taste ... frog

how about oily?

also make sure to put amitin in their meat. mushrooms have it.